intro-to-object-oriented-programming/defining-classes.md

```diff
@@ -0,0 +1,495 @@
+
+## When Do We Make a Class?
+
+Classes represent _things_, _ideas_, _concepts_, or _other kinds of nouns_ that are relevant to the problem at hand.
+
+Developers define classes after realizing that encapsulating (organizing) state and behavior into one data type will make problem-solving easier.
+
+Developers define classes any time:
+
+- At the beginning of a project, after reading a problem statement
+- In the middle of a project, after discovering that a deeper part of the problem needs a new class
+- While refactoring, if seeing that attributes and methods can be better organized as a class
+
+Some considerations for making a class:
+
+- Is there a _concept_ that is used frequently in this project?
+- Is there a major concept that holds a bunch of data?
+- Is there a major concept that has certain behaviors?
+- Will defining a class increase the readability?
+- Will defining a class organize the code better?
+- Can this class be reused in the future?
+
+Classes can be extremely big or extremely small; there is no right size for a class.
+
+Classes can also follow the [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle). Before we create a class, we should do our best to determine:
+
+1. What idea/concept is this class responsible for representing?
+1. What is the best name for this class?
+   - The name should be singular, ie `Album` _not_ Albums
+
+## Defining Classes
+
+Once we've determined a class name and its responsibility, we can define a class with this syntax:
+
+```python
+class ClassName:
+    pass
+```
+
+| Piece of Code | Notes                                                                                                                                                                               |
+| ------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `class`       | A keyword that indicates that a class definition is beginning                                                                                                                       |
+| `ClassName`   | **Replace this** with the name of the class you're defining. The PEP-8 style guide prefers CapitalCamel case for class names. Class names should be singular nouns.                 |
+| `:`           | Don't forget the colon to indicate the beginning of the class body                                                                                                                  |
+| Class body    | Code that describes the class definition. It should be indented once into the class. The class body will include attributes and methods (functions) to describe state and behavior. |
```

Comment:
About `pass`: Nowhere in the current C15 curriculum yet. Instead of forcing a paragraph about `pass` somewhere where I don't feel confident, I wanna wait out and see if students **and instructors ;) ** can get through that hurdle through means other than the written curriculum. For a future iteration I hope to have better data.

This may seem like overengineering for one keyword that can be defined in one, maybe two, MAYBE three sentences, but idk, I'm pretty curious.

(AKA I think that dedicating a whole paragraph to `pass` out of context, where a student will find the need to context-switch and then re-read the paragraph, look up an example, consider why it's useful (how do you describe writing stubs to a newbie and make it worth their time) and then commit it to memory which might take up to 15 minutes, when I'm curious about finding a way to make that... 2 minutes. And my current idea is that going "oh idk what `pass` is" could be one way to make it 2 minutes)